SwiftGit2/SwiftGit2.swift

```diff
@@ -0,0 +1,14 @@
+//
+//  SwiftGit2.swift
+//  SwiftGit2
+//
+//  Created by Andrew Breckenridge on 10/23/17.
+//  MIT.
+//
+
+import libgit2
+
+// swiftlint:disable:next identifier_name
+static func SwiftGit2Init(_: Void) {
```

Comment:
This isn't equivalent to what was in the Obj-C version. The `__attribute__((constructor))` causes the function to run when the library is loaded. We'd need some sort of equivalent mechanism here. (It could probably be replaced with a well-placed `dispatch_once`.)

Reply:
Great catch. In swift this should just be possible with a `static let`, I remember reading that it accomplishes the same result as the `dispatch_once` dance in Obj-C. Can we write a test for this?

Reply:
Looking at the documentation for `git_libgit2_init`, we might be able to just call this in `init` and `dealloc` of `Repository`.

Reply:
I'm not sure how to write a test for it. CI wasn't able to build with your changes. Do the tests successfully run without calling this?

Reply:
I wasn't sure either, let's circle back once CI for SwiftPM is up